### PR TITLE
spec updates

### PR DESCRIPTION
**Pull request by arnetheduck**
* balances move out to separate seq
* bunch of placeholders for proof-of-custody / phase1

### CODE REVIEW

File: beacon_chain/state_transition.nim [hunk 1]
```diff
@@ -418,6 +413,18 @@ func lowerThan(candidate, current: Eth2Digest): bool =
     if v > candidate.data[i]: return true
   return false
 
+func inclusion_slot(state: BeaconState, v: Uint24): uint64 =
+  for a in state.latest_attestations:
+    if v in get_attestation_participants(state, a.data, a.participation_bitfield):
+      return a.slot_included
+  doAssert false # shouldn't happen..
```
Comment (mratsim) [inline beacon_chain/state_transition.nim]: Maybe we should create an `unreachable` template?

File: beacon_chain/state_transition.nim [hunk 2]
```diff
@@ -612,76 +611,80 @@ func processEpoch(state: var BeaconState) =
...
+            inactivity_penalty(state, index, epochs_since_finality)
+        if index notin previous_epoch_boundary_attester_indices:
+          state.validator_balances[index] -=
+            inactivity_penalty(state, index, epochs_since_finality)
+        if index notin previous_epoch_head_attester_indices:
+          state.validator_balances[index] -=
+            inactivity_penalty(state, index, epochs_since_finality)
+        if state.validator_registry[index].status == EXITED_WITH_PENALTY:
+          state.validator_balances[index] -=
+            3'u64 * inactivity_penalty(state, index, epochs_since_finality)
```
Comment (mratsim) [inline beacon_chain/state_transition.nim]: Can all of this ever go into "negative" and rollover to very high balances?

File: beacon_chain/state_transition.nim (hunk 1 above)
Comment (arnetheduck) [reply beacon_chain/state_transition.nim]: In this particular case, I think it should be possible to rewrite the code to eliminate the loop completely - in other words `shouldn't happen` can be turned into `can't happen` which I prefer as an overall approach.. the reason it looks like this is.. yes! the spec!

File: beacon_chain/state_transition.nim (hunk 2 above)
Comment (arnetheduck) [reply beacon_chain/state_transition.nim]: it "shouldn't" - there's a safety margin on the ETH where the validator gets ejected well before it's even close.. but in general, I agree, we need to make a pass over the code and work on these over- and underflow scenarios generally. it's noted in a TODO somewhere.

**Pull request merged**